homeassistant/components/surepetcare/__init__.py

```diff
@@ -198,23 +198,22 @@ async def handle_set_lock_state(self, call: ServiceCall) -> None:
         await self.lock_states[state](flap_id)
         await self.async_request_refresh()
 
-    def get_pets(self) -> dict[int, str]:
+    def get_pets(self) -> dict[str, int]:
         """Get pets."""
-        names = {}
+        pets = {}
         for surepy_entity in self.data.values():
             if surepy_entity.type == EntityType.PET and surepy_entity.name:
-                names[surepy_entity.id] = surepy_entity.name
-        return names
+                pets[surepy_entity.name] = surepy_entity.id
+        return pets
 
     async def handle_set_pet_location(self, call: ServiceCall) -> None:
         """Call when setting the pet location."""
         pet_name = call.data[ATTR_PET_NAME]
         location = call.data[ATTR_LOCATION]
-        for device_id, device_name in self.get_pets().items():
-            if pet_name == device_name:
-                await self.surepy.sac.set_pet_location(
-                    device_id, Location[location.upper()]
-                )
-                await self.async_request_refresh()
-                return
-        _LOGGER.error("Unknown pet %s", pet_name)
+        try:
+            device_id = self.get_pets()[pet_name]
+        except ValueError:
```

Comment:
I meant that we should raise `ValueError` here if the pet name is missing to notify the user in the frontend about the error. Raise an exception in a service handler will do that.

Reply:
Ahh, I did not know. Thanks

Reply:
We don't even need to raise an error I see now since we validate the name in the service validation schema already.